client/src/layouts/Events/RegisterEventForm/index.tsx

```diff
@@ -0,0 +1,396 @@
+    },
+    {
+        title: "WA",
+        id: "WA"
+    },
+    {
+        title: "WV",
+        id: "WV"
+    },
+    {
+        title: "WI",
+        id: "WI"
+    },
+    {
+        title: "WY",
+        id: "WY"
+    }
+]
+
+const initValues:FormFields = { 
+    name: '',
+    category: '',
+    date: '',
+    starttime: '',
+    endtime: '',
+    description: '',
+    provider: '',
+    url: '',
+    location: '',
+    street: '',
+    city: '',
+    state: '',
+    zipcode: '',
+}
+
+export default function RegisterEventForm(): React.ReactElement {
+
+    const [img64Base, setImg64Base] = React.useState<string | undefined>(undefined)
+    //const [inputFileRef, setInputFileRef] = React.useState<React.RefObject<HTMLInputElement> | null>(null)
+
+    const [wholeDayEvent, setWholeDayEvent] = useState(false)
+    const [onlineEvent, setOnlineEvent] = useState(false)
+
+    const methods = useForm<FormFields>({
+        defaultValues: initValues,
+        resolver: yupResolver(EventsSchema), 
+        mode: "onChange"
+    });
+
+    const onSubmit = (data:FormFields) => {
+
+        Swal.fire({
+            title: "We have received your event!",
+            text: "After our team reviews the information, we'll make sure your event gets the visibility it deserves.",
+            icon: "success"
+        });
+
+        console.log(data)
+    }
+    
+    const handleWholeDayEvent = () => setWholeDayEvent(!wholeDayEvent)
+    const handleOnlineEvent = () => setOnlineEvent(!onlineEvent)
+
+    return (
+        <article className={styles.formContainer}>
+            <FormProvider {...methods}>
+                <form className={styles.eventForm} onSubmit={methods.handleSubmit(onSubmit)}>
+                    <div className={styles.eventDetailsContainer}>
+                        <EventFormTitle title="Event Details" />
+                        <TextFieldBase id="name" title='Event name*' placeholder="Event name" fullwidth={true} />
+                        <TextFieldBase id="provider" title='Event provider*' placeholder="Event provider" fullwidth={true} />
+                        <div className={styles.inputGridContainer}>
+                            <div className={styles.gridFullColumn} style={{ gridColumn: 'span 6' }}>
+                                <TextFieldBase id="date" title='Event date*' placeholder="Event date" type="date" fullwidth={true} />
+                            </div>
+                            <div className={styles.gridHalfColumn} style={{ gridColumn: 'span 3' }}>
+                                <TextFieldBase id="starttime" disabled={wholeDayEvent} title='Start time' placeholder="HH:MM" type="time" fullwidth={true} />
+                            </div>
+                            <div className={styles.gridHalfColumn} style={{ gridColumn: 'span 3' }}>
+                                <TextFieldBase id="endtime" disabled={wholeDayEvent} title='End time' placeholder="HH:MM" type="time" fullwidth={true} />
```

Comment:
We can have two sections - a "Start date" section in which they include the date and time that the event starts. And below that we can have an "End date" section for the date and time when the event ends. 

This would eliminate the need to have a "wholeDayEvent" checkbox and will also allow users to submit multi-day events such as conferences.